### PR TITLE
CORE-1876 Upgrade to next.js v12

### PR DESCRIPTION
**Pull request by psarando**
This PR will upgrade the UI to `next.js` v12, as well as `next-i18next` to v13, and `react` to v17.

This required some updates to the way `i18n` is configured for all of our `next.js` pages, as well as the `I18nProviderWrapper` for `Storybook` and `Jest` tests.

There should not be any visible changes to the UI, but one of the big changes in this upgrade is that translation namespaces are loaded server-side, so the client should not have to download as many JSON files for i18n messages.

### CODE REVIEW

File: next-i18next.config.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+const path = require("path");
+
+const isProduction = process.env.NODE_ENV === "production";
+const localePath =
+    typeof window === "undefined"
+        ? path.resolve("./public/static/locales")
+        : "/static/locales";
+
+module.exports = {
+    debug: !isProduction,
```
Comment (psarando) [inline next-i18next.config.js:10]: Turning on the debug flag in dev helps catch i18n key typos and missing i18n messages, which already helped me catch these: 8415b299a226ab229030dbe378e4c2e037e005a8

File: src/components/data/SelectionDrawer.js [hunk 2]
```diff
@@ -155,29 +155,30 @@ function SelectionToolbar(props) {
...
-                                    : theme.spacing(1),
-                            }}
-                            size="small"
-                        >
-                            {t("selCurrentFolder")}
-                        </Button>
-                    </>
-                )}
+                {acceptedType === ResourceTypes.FOLDER &&
+                    selectedTotal === 0 && (
```
Comment (psarando) [inline src/components/data/SelectionDrawer.js:159]: My editor wanted to format this section with prettier 🤷

File: src/__mocks__/i18nProviderWrapper.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+ * @returns {JSX.Element}
+ * @constructor
+ */
+export function I18nProviderWrapper({ children }) {
+    return (
+        <Suspense fallback={"Loading i18n..."}>
+            <I18nextProvider i18n={I18NextClient}>{children}</I18nextProvider>
+        </Suspense>
+    );
+}
```
Comment (slr71) [inline src/__mocks__/i18nProviderWrapper.js:35]: 👍

File: public/static/locales/en/common.json [hunk 4]
```diff
@@ -90,6 +90,7 @@
     "update": "Update",
     "updateAccountInformation": "Update Account Information",
     "updateNow": "Update Now",
+    "usageSummaryError": "Unable to get usage summary.",
```
Comment (psarando) [inline public/static/locales/en/common.json:93]: Moving this message to the `common` ns prevents other pages from requiring the entire `dashboard` ns for only this message.

File: src/components/dashboard/dashboardItem/ResourceUsageItem.js [hunk 5]
```diff
@@ -71,7 +70,7 @@ export default function ResourceUsageItem(props) {
             />
             {resourceUsageError && (
                 <ErrorTypographyWithDialog
-                    errorMessage={t("usageSummaryError")}
+                    errorMessage={t("common:usageSummaryError")}
```
Comment (slr71) [inline src/components/dashboard/dashboardItem/ResourceUsageItem.js:73]: 👍

File: src/pages/dashboard.js [hunk 6]
```diff
@@ -1,13 +1,25 @@
...
-DashboardPage.getInitialProps = async () => ({
-    namespacesRequired: ["dashboard", "apps"],
-});
+export async function getStaticProps(context) {
+    // The `locale` prop is undefined in this page for some reason.
+    const { locale, defaultLocale } = context;
+
+    return {
+        props: {
+            ...(await serverSideTranslations(locale ?? defaultLocale, [
```
Comment (psarando) [inline src/pages/dashboard.js:16]: The `locale` prop is undefined in this page for some reason (but not the `defaultLocale`). I think due to the way we render this page for the root path:
https://github.com/cyverse-de/sonora/blob/961672814c603ce020b46293e70e44985b7882ec/src/server/index.js#L174-L176

**Review by slr71 — APPROVED** [review]
Looks good to me! :thu

**Comment by psarando** [comment]
Thanks for the review! 👍 

I'll go ahead and merge now, but additional reviews still welcome 🙇

**Pull request merged**